yw_website/yw_website/apps/website/serializers.py

```diff
@@ -61,3 +66,102 @@ class TagFileSerializer(serializers.ModelSerializer):
     class Meta:
         model = TagFile
         fields = '__all__'
+
+class YesWorkflowSaveSerializer(serializers.ModelSerializer):
+    model = serializers.CharField(required=False, allow_blank=False)
+    model_checksum = serializers.CharField(required=False, allow_blank=True, max_length=128)
+    graph = serializers.CharField(required=False, allow_blank=False)
+    recon = serializers.CharField(required=False, allow_blank=True)
+    tags = serializers.ListField(
+        child=serializers.CharField(required=False, allow_blank=True, max_length=32)
+    )
+    scripts = ScriptSerializer(many=True)
+    files = FileSerializer(many=True)
+
+    class Meta:
+        model = Workflow
+        fields =('title','description','model','model_checksum', 'graph', 'recon', 'tags', 'scripts', 'files' )
+
+
+    def create(self, validated_data):
+        w = Workflow(
+            user=validated_data.get('username'),
+            title=validated_data.get('title'),
+            description=validated_data.get('description')
+        )
+        w.save()
+        v = Version(
+            workflow=w,
+            yw_model_checksum=validated_data.get('model_checksum'),
+            yw_model_output=validated_data.get('model'),
+            yw_graph_output=validated_data.get('graph'),
+            last_modified=datetime.datetime.now(tz=timezone.utc)
+        )
+        v.save()
+        r = Run(
+            version=v,
+            run_time_stamp=datetime.datetime.now(tz=timezone.utc),
+            yw_recon_output=validated_data.get('recon')
+        )
+        r.save()
+
+        tags = TagSerializer(validated_data.get('tags'), many=True)
+        for tag in tags.data:
+            t = Tag(parent_tag=None, tag_type=Workflow, title=tag.get('title'))
+            t.save()
+
+        scripts = ScriptSerializer(validated_data.get('scripts'), many=True)
```

Comment:
can we make sure that there is at least one script as well? like if scripts.data is `None` or whatever that may look like, throw an error that signifies we want to return a `500` status code

Reply:
If it is required than the serializer will auto return an error

Reply:
lets say the request has something like this in its body
```
{
    . . .
    'scripts': []
    . . .
}
```
Would the serializer auto return an error? if not, I feel that we should be verifying that there is at least once script, since it would not be a valid run if no script was run.

Reply:
Done